content/Updating-musl-1.2.5.md

```diff
@@ -0,0 +1,107 @@
++++
+path = "2025/07/01/Updating-musl-1.2.5"
+title = "Updating Rust's Linux musl targets to 1.2.5"
+authors = ["Aria Desires"]
+description = "musl targets will soon ship with musl 1.2.5"
+
+[extra]
+team = "The Compiler Team"
+team_url = "https://www.rust-lang.org/governance/teams/compiler"
++++
+
+# Updating Rust's Linux musl targets to 1.2.5
+
+Beginning with Rust 1.93 (slated for stable release on 2026-01-22), the various `*-linux-musl` targets will all [ship](https://github.com/rust-lang/rust/pull/142682) with musl 1.2.5. This primarily affects static musl builds for `x86_64`, `aarch64`, and `powerpc64le` which bundled musl 1.2.3. This update comes with [several fixes and improvements](https://musl.libc.org/releases.html), and a breaking change that affects the Rust ecosystem.
+
+For the Rust ecosystem, the primary motivation for this update is to receive major improvements to musl's DNS resolver which shipped in 1.2.4 and received bug fixes in 1.2.5. When using `musl` targets for static linking, this should make portable linux binaries that do networking more reliable, particularly in the face of large DNS records and recursive nameservers.
+
+However, 1.2.4 also comes with a breaking change: [the removal of several legacy compatibility symbols that the Rust libc crate was using](https://github.com/rust-lang/libc/issues/2934). A fix for this [was shipped in libc 0.2.146 in June 2023 (2 years ago)](https://github.com/rust-lang/libc/pull/2935), and we have been waiting for newer versions of the libc crate to propagate throughout the ecosystem before shipping the musl update.
+
+A crater run in July 2024 found only about 2.4% of Rust projects were still affected. [A crater run in June 2025 found 1.5% of Rust projects were affected](https://github.com/rust-lang/rust/pull/142682#issuecomment-3002514461). Most of that change is from crater analyzing More Rust Projects. The absolute amount of broken projects went down by 15% while the absolute amount of analyzed projects went up by 35%.
+
+At this point we expect there will be minimal breakage, and most breakage should be resolved by a `cargo update`. We believe this update shouldn't be held back any longer, as it contains critical fixes for the musl target.
+
+Manual inspection of some of the affected projects indicates they largely haven't run `cargo update` in 2 years, often because they haven't had any changes in 2 years. Analysis suggests this Rust release will be a devastating blow to the thriving ecosystem of Rust gamedev projects that you started 3 years ago and will definitely get back to this year.
```

Comment:
Suggestion: this sentence comes across as a bit snarky to / dismissive of the authors of those gamedev projects. I would maybe just say something more neutral

> Analysis of impacted projects suggests this Rust release will affect quite a few gamedev projects that have not run `cargo update` in several years.